app/models/meeting.rb

```diff
@@ -1,4 +1,25 @@
 class Meeting < ActiveRecord::Base
   belongs_to :course_instance
   belongs_to :professor
+
+  MEETING_TIME_REGEX = /\d{1,2}:\d{1,2} [AP]M/
+
+  # Return an Array of ScheduledMeetings for this Meeting.
+  # Empty if the schedule is TBA
+  # 
+  # All schedules have the form:
+  #   DAY_ABBRVS START_TIME - END_TIME
+  def scheduled_meetings
+    return [] if schedule.start_with?('TBA') # sometimes there's trailing whitespace
+
+    # the start time is the first time in the schedule
+    start_time = schedule.match(/^.* (#{MEETING_TIME_REGEX}) - #{MEETING_TIME_REGEX}$/).captures.first
+    # the end time is the second time
+    end_time = schedule.match(/^.* #{MEETING_TIME_REGEX} - (#{MEETING_TIME_REGEX})$/).captures.first
```

Comment:
Also, FYI, you usually won't need to use `.captures`. Even in your case you could have just replaced the `.captures.first` with `[1]`